src/Autofac.Pooling/DefaultPooledRegistrationPolicy.cs

```diff
@@ -0,0 +1,56 @@
+﻿using System;
+using System.Collections.Generic;
+using Autofac.Core;
+
+namespace Autofac.Pooling
+{
+    /// <summary>
+    /// Provides the default <see cref="IPooledRegistrationPolicy{TPooledObject}"/>.
+    /// </summary>
+    /// <typeparam name="TPooledObject">The type of object being pooled.</typeparam>
+    public class DefaultPooledRegistrationPolicy<TPooledObject> : IPooledRegistrationPolicy<TPooledObject>
+        where TPooledObject : class
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="DefaultPooledRegistrationPolicy{TLimit}"/> class.
+        /// </summary>
+        public DefaultPooledRegistrationPolicy()
+        {
+            MaximumRetained = Environment.ProcessorCount * 2;
```

Comment:
What was the logic behind this being the default? (I mean, it's a fine default, I don't have better suggestions, just curious.) Might be good to have a `// comment` explaining the logic so we know later if it's safe to refactor or if there's something deeper behind it.

Reply:
That is how the default size is calculated in the .NET Runtime code for the ObjectPool, so I just re-used it.

Added a comment to that effect.